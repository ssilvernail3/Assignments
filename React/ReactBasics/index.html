<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Shiba Inu</title>
</head>

<body>
    <h1>SHIBAS!!</h1>
    <div id="root"></div>
    <script src="https://unpkg.com/react/umd/react.development.js"></script>
    <script src="https://unpkg.com/react-dom/umd/react-dom.development.js"></script>
    <script src="https://unpkg.com/babel-standalone"></script>
    
    <script src="Shiba.js" type="text/jsx"></script>
    <script src="Sharpei.js" type="text/jsx"></script>
    <script src="RandonNum.js" type="text/jsx"></script>
    <script src="Animal.js" type="text/jsx"></script>
    <script src="App.js" type="text/jsx"></script>
</body>

</html>