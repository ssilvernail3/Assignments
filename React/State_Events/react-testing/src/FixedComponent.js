import React from "react";

function FixedComponent(props) {
    return (
        <div>
            <p>Hello! I am a FixedComponent!</p>
            <p>Here are some numbers: </p>
            <p>{ props.favNum }</p>
            <p>{ props.favNum + 1}</p>
            <p>{ props.favNum + 2}</p>
            <p>{ props.favNum + 3}</p>
        </div>
    )
}

FixedComponent.defaultProps = {
    favNum: 42
};

export default FixedComponent