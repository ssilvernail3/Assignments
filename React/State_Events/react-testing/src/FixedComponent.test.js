import React from 'react';
import { render, screen } from '@testing-library/react';
import FixedComponent from './FixedComponent';

test('if renders without crahsing', () => {
  render(<FixedComponent />)
}); 

it('matches snapshot',  () => {
    const { asFragment } = render(<FixedComponent />);
    expect(asFragment()).toMatchSnapshot(); 
})