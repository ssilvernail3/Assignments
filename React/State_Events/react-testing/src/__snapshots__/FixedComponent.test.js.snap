// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      Hello! I am a FixedComponent!
    </p>
    <p>
      Here are some numbers: 
    </p>
    <p>
      42
    </p>
    <p>
      43
    </p>
    <p>
      44
    </p>
    <p>
      45
    </p>
  </div>
</DocumentFragment>
`;
